llvm-spirv/test/transcoding/clk_event_t.cl

```diff
@@ -2,11 +2,13 @@
 // RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
 // RUN: llvm-spirv %t.bc -o %t.spv
 // RUN: spirv-val %t.spv
-// RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+// RUN: llvm-spirv -r -opaque-pointers=1 %t.spv -o %t.rev.bc
 // RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM-OCL
-// RUN: llvm-spirv -r --spirv-target-env=SPV-IR %t.spv -o %t.rev.bc
+// RUN: llvm-spirv -r -opaque-pointers=1 --spirv-target-env=SPV-IR %t.spv -o %t.rev.bc
 // RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM-SPV
 // RUN: llvm-spirv %t.rev.bc -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+//
+// XFAIL: *
```

Comment:
This test still uses typed pointers, so I think we can use either XFAIL or `-opaque-pointers=1`, but not both of them at the same time.
BTW, the right name of the `llvm-spirv` option is `emit-opaque-pointers`. We should update CHECKs to use opaque pointers ASAP!

Reply:
IMO the test or translator is broken. It should generate target extension type, but it generates ptr. Tagging @jcranmer-intel

Reply:
The test is definitely wrong - it doesn't check target extension type.
From your message it looks like another bug is in the clang.